### ISSUE TITLE
Improve issue matching with branch names

### ISSUE THREAD

**Issue by justinrobots** [gitkraken/vscode-gitlens#3894]
The branch for this issue is `release-notes-16-1` and GitLens thinks these two issues (`#16` and `#1`) are associated but they are not. I suspect the matching algorithm needs to be refined to avoid mismatches like this.
 
![Image](https://github.com/user-attachments/assets/4704728c-bb19-4c10-b920-94f191066b9f)

**Comment by nzaytsev** [comment]
I've made some improvements with the **non-prefixed** autolinks.
- Now I'm trying to consider issue title and its' matched numbers to remove them from the final list of the autolinks
- I'm searching for number chunks no longer than 2 numbers
- I'm trying to ignore the branch name chunks describing the release version as much as it's possible

You can see how it works in the [new test group ](https://github.com/gitkraken/vscode-gitlens/pull/3898/files#diff-cb9870e106f1dc0205d5f189d04da778272b3449dc795404efc665861ea7c55aR61-R130)

I actually don't like the way it's done, it's a bit anti-pattern, but now it works slightly more precisely with **non-prefixed** keys. I'm not sure if it won't make the detection slower. And it still can not work with 100% precise. I described at least a few cases (in the test file) which we can not definitely associate with an issue

In my opinion, we should have a button to de-associate the mismatched issues with the branch. No matter how hard we try to "tune" the autolinks algorithm, the patterns of branch name writing vary greatly from team to team and from user to user.

As @sergeibbb suggests, we could add an additional setting for the autolinks that would disable the basic algorithm and allow the user to use their own pattern

**Comment by justinrobots** [comment]
>In my opinion, we should have a button to de-associate the mismatched issues with the branch. No matter how hard we try to "tune" the autolinks algorithm, the patterns of branch name writing vary greatly from team to team and from user to user.

I agree that we should find a good way to allow users to disassociate issues that are inaccurately matched.

**Comment by nzaytsev** [comment]
@justinrobots - I tried to do that in scope of the new task #3956

**Comment by sergeibbb** [comment]
@justinrobots @eamodio @axosoft-ramint 

Testing results:

![Image](https://github.com/user-attachments/assets/5d466aa0-b377-4131-9acb-14f0653f15c7)
